tick/stateful/evaluation_funcs.go

```diff
@@ -0,0 +1,789 @@
+package stateful
+
+import (
+	"regexp"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+type operationKey struct {
+	operator  tick.TokenType
+	leftType  ValueType
+	rightType ValueType
+}
+
+var boolTrueResultContainer = resultContainer{BoolValue: true, IsBoolValue: true}
+var boolFalseResultContainer = resultContainer{BoolValue: false, IsBoolValue: true}
+var emptyResultContainer = resultContainer{}
+
+var evaluationFuncs = map[operationKey]evaluationFn{
+	// -----------------------------------------
+	//	Comparison evaluation funcs
+
+	operationKey{operator: tick.TokenAnd, leftType: TBool, rightType: TBool}: func(scope *tick.Scope, executionState ExecutionState, leftNode, rightNode NodeEvaluator) (resultContainer, *ErrSide) {
+		var left bool
+		var right bool
+		var err error
+
+		if left, err = leftNode.EvalBool(scope, executionState); err != nil {
+			return boolFalseResultContainer, &ErrSide{error: err, IsLeftSide: true}
+		}
+
+		// Short circuit evaluation
```

Comment:
This reminds me: Can you add a changelog entry? Note that short circuiting evaluation is a breaking change. Must of time people won't care but if someone did something like this, the behavior has changed.

```
lambda: "bool_value" && (count() > 100)
```

Now `count` will only be called with `bool_value` is true.